runtime/vm/jfr.cpp

```diff
@@ -1125,21 +1125,30 @@ jfrSamplingThreadProc(void *entryArg)
 }
 
 jboolean
-setJFRRecordingFileName(J9JavaVM *vm, char *fileName)
+setJFRRecordingFileName(J9JavaVM *vm, char *newFileName)
 {
-	if (NULL != vm->jfrState.jfrFileName) {
-		UDATA defaultFileNameLen = strlen(DEFAULT_JFR_FILE_NAME);
-		const char *jfrFileName = vm->jfrState.jfrFileName;
-		if ((defaultFileNameLen != strlen(jfrFileName))
-				|| (0 != strncmp(DEFAULT_JFR_FILE_NAME, jfrFileName, defaultFileNameLen))
-		) {
-			PORT_ACCESS_FROM_JAVAVM(vm);
-			j9mem_free_memory(vm->jfrState.jfrFileName);
+	const char *existingFileName = vm->jfrState.jfrFileName;
+	UDATA existingfileNameLength = strlen(existingFileName);
+	jboolean res = JNI_TRUE;
+
+	if ((strlen(newFileName) != existingfileNameLength)
+		|| (0 != strncmp(existingFileName, newFileName, existingfileNameLength))
+	) {
```

Comment:
This doesn't solve the problem of leaking if `newFileName` is equal to `DEFAULT_JFR_FILE_NAME` or the previously specified name. In fact, this is worse in that it removes the check that `vm->jfrState.jfrFileName` is not null before calling `strlen()`.

I suggest want a better fix than #20988.